File: 1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        subMap = dict()
        for i, val in enumerate(nums):
            if subMap.get(val) is not None:
                return [subMap.get(val), i]
            subMap[target-val] = i
# @lc code=end

File: 1.两数之和.rs
/*
 * @lc app=leetcode.cn id=1 lang=rust
 *
 * [1] 两数之和
 */

// @lc code=start
use std::collections::HashMap;
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
         let mut subMap = HashMap::new();
         for (i, &x) in nums.iter().enumerate() {
            let y = target - x;
            if let Some(&j) = subMap.get(&x) {
                return vec![j as i32, i as i32];
            }
            subMap.insert(y, i as i32);
         }
        return vec![]
    }
}
// @lc code=end

